spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/function/KafkaStreamsBindableProxyFactory.java

```diff
@@ -165,12 +183,30 @@ private boolean isAnotherFunctionOrConsumerFound(ResolvableType arg1) {
 		int numberOfInputs = this.type.getRawClass() != null &&
 				(this.type.getRawClass().isAssignableFrom(BiFunction.class) ||
 						this.type.getRawClass().isAssignableFrom(BiConsumer.class)) ? 2 : getNumberOfInputs();
+
+		// For @Component style beans.
+		if (method != null) {
+			final ResolvableType returnType = ResolvableType.forMethodReturnType(method);
+			final Object bean = beanFactory.getBean(functionName);
```

Comment:
This can result in exception if bean not found. Consider using `beanFactory.containsBean`